### ISSUE TITLE
Private profiles do not work

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#808]
RPCAPI will ignore private profiles. There is no code to check the profile in `start_domain_test` against the list of private profiles.

Scenario 1:
```
[PUBLIC PROFILES]
default-no-ipv6=/usr/local/etc/zonemaster/profile-no-ipv6.json
default=/usr/local/etc/zonemaster/profile.json
test=/usr/local/etc/zonemaster/profile-test1.json

[PRIVATE PROFILES]
#test=/usr/local/etc/zonemaster/profile-test1.json
```
```
# zmb --verbose start_domain_test --domain sunet.se --profile test
{"id":1,"method":"start_domain_test","jsonrpc":"2.0","params":{"profile":"test","domain":"sunet.se"}}
{"result":"5d94caf2d74d76f1","id":1,"jsonrpc":"2.0"}
```
Scenario 2:
```
[PUBLIC PROFILES]
default-no-ipv6=/usr/local/etc/zonemaster/profile-no-ipv6.json
default=/usr/local/etc/zonemaster/profile.json
#test=/usr/local/etc/zonemaster/profile-test1.json

[PRIVATE PROFILES]
test=/usr/local/etc/zonemaster/profile-test1.json
```
```
# zmb --verbose start_domain_test --domain sunet.se --profile test
{"params":{"profile":"test","domain":"sunet.se"},"jsonrpc":"2.0","id":1,"method":"start_domain_test"}
{"jsonrpc":"2.0","error":{"code":-32603,"message":"Internal error 009 \n"},"id":1}
```

**Comment by mattias-p** [comment]
I find it curious that nobody has reported this before. It seems to me nobody is using the private profiles feature.

**Comment by matsduf** [comment]
Will be resolved by #808

**Comment by matsduf** [comment]
> I find it curious that nobody has reported this before. It seems to me nobody is using the private profiles feature.

@mattias-p, I agree. Is there really a real use case for it? Or should we do a direct deprecation/removal of the feature?

**Comment by mattias-p** [comment]
I think it's nicer to deprecate it first. It'd also make sense to deprecate PUBLIC PROFILES and introduce PROFILES as a replacement.

**Comment by matsduf** [comment]
Since it has never worked, the deprecation here should be to warn if there is any "private profiles" section in the configuration. There is no reason to make private profiles work.

**Comment by matsduf** [comment]
Resolved by #801
